addons/l10n_in_edi/models/res_company.py

```diff
@@ -82,16 +84,57 @@ def _l10n_in_edi_authenticate(self):
     def _l10n_in_check_einvoice_validation(self):
         checks = {
             'company_address_missing': {
-                'fields': ('street', 'zip', 'city', 'state_id', 'country_id',),
-                'message': _("Companies should have a complete address, verify their Street, City, State, Country and Zip code."),
+                'message': _(
+                    "Companies should have a complete address, verify their Street, City, State, "
+                    "Country and Zip code."
+                ),
+                'validation': lambda record: any(
+                    not record[field] for field in (
+                        'street', 'zip', 'city', 'state_id', 'country_id',
+                    )
+                ),
+            },
+            'company_email_check': {
+                'message': _(
+                    "Email address should be valid and not more then 100 characters."
+                ),
+                'validation': lambda record: (
+                    record.email and (
+                        not re.match(
+                            r"^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$", record.email
+                        ) or not re.match("^.{6,100}$", record.email)
+                    )
+                ),
+            },
+            'company_phone_check': {
+                'message': _(
+                    "Mobile number should be minimum 10 or maximum 12 digits."
+                ),
+                'validation': lambda record: (
+                    record.phone and not re.match(
+                        r"^[0-9]{10,12}$",
+                        self.env['account.move']._l10n_in_extract_digits(record.phone)
+                    )
+                ),
+            },
+            'company_street2_check': {
+                'message': _(
+                    "Street2 should be min 3 and max 100 characters"
+                ),
+                'validation': lambda record: (
+                    record.street2 and not re.match("^.{3,100}$", record.street2)
+                ),
             },
         }
         return {
             f"l10n_in_edi_{key}": {
                 'message': check['message'],
-                'action_text': _("View Company/ies"),
+                'action_text': (
+                    _("View Companies") if len(invalid_records) > 1
+                    else _("View Company: %s" % invalid_records.name)
```

Comment:
```suggestion
                    else _("View %s", invalid_records.name)
```
When using `_` formatted text must passed as parameter